core/src/Revolution/Processors/Context/GetList.php

```diff
@@ -79,7 +84,28 @@ public function prepareQueryBeforeCount(xPDOQuery $c)
                 'key:NOT IN' => is_string($exclude) ? explode(',', $exclude) : $exclude,
             ]);
         }
-
+        /*
+            When this class is used to fetch data for a grid filter's store (combo),
+            limit results to only those contexts present in the current grid.
+        */
+        if ($this->isGridFilter) {
+            if ($userGroup = $this->getProperty('usergroup', false)) {
+                $c->innerJoin(
+                    modAccessContext::class,
+                    'modAccessContext',
+                    [
+                        '`modAccessContext`.`target` = `modContext`.`key`',
+                        '`modAccessContext`.`principal` = ' . (int)$userGroup,
```

Comment:
For what cases is explicit type casting here?

Reply:
Although the usergroup property _should_ only be exposed internally, casting it to int ensures only a safe value will be sent in the query. Note that this format of the join's conditions (a non-associative array of strings) does not get sanitized.